### PR TITLE
fix(protocol): reward non-assigned prover 7/8 liveness bond

### PR DESCRIPTION
**Pull request by dantaik**
Without this small fix, provers other than the assigned prover won't have an incentive to prove a block out of the proving window.

### CODE REVIEW

**Review by davidtaikocha — APPROVED** [review]

**Review by davidtaikocha — APPROVED** [review]

File: packages/protocol/contracts/layer1/based/LibProving.sol [hunk 1]
```diff
@@ -607,10 +608,17 @@ library LibProving {
...
                 if (_returnLivenessBond(_local, _proof.data)) {
                     if (_local.assignedProver == msg.sender) {
-                        reward += _local.livenessBond;
+                        unchecked {
+                            reward += _local.livenessBond;
+                        }
                     } else {
                         LibBonds.creditBond(_state, _local.assignedProver, _local.livenessBond);
                     }
+                } else if (!_local.isTopTier) {
```
Comment (dantaik) [inline packages/protocol/contracts/layer1/based/LibProving.sol:617]:
```suggestion
                } else {
```

File: packages/protocol/contracts/layer1/based/LibProving.sol [hunk 2]
```diff
@@ -607,10 +608,17 @@ library LibProving {
...
                     if (_local.assignedProver == msg.sender) {
-                        reward += _local.livenessBond;
+                        unchecked {
+                            reward += _local.livenessBond;
+                        }
                     } else {
                         LibBonds.creditBond(_state, _local.assignedProver, _local.livenessBond);
                     }
+                } else {
+                    // Reward 50% of liveness bond to the actual prover, the other 50% burnt.
```
Comment (adaki2004) [inline packages/protocol/contracts/layer1/based/LibProving.sol:618]: I think the comment is incorrect here. Example: 100 * 7 / 8 = 87.5% not 50%.

**Review by Brechtpd — APPROVED** [review]

**Comment by adaki2004** [comment]
Also since the proposer is paying the liveness bonds (in LibProposing):
`LibBonds.debitBond(_state, _resolver, local.params.proposer, _config.livenessBond);`

Should not we take into consideration the proposer when it comes to returning it ? Maybe it changed in the meantime, but i was in the belief that prover pays the bond, not the proposer. 🤔  (Or proposer == prover ? but i dont see this enforced anywhere )

**Comment by dantaik** [comment]
> Also since the proposer is paying the liveness bonds (in LibProposing): `LibBonds.debitBond(_state, _resolver, local.params.proposer, _config.livenessBond);`
> 
> Should not we take into consideration the proposer when it comes to returning it ? Maybe it changed in the meantime, but i was in the belief that prover pays the bond, not the proposer. 🤔 (Or proposer == prover ? but i dont see this enforced anywhere )

The proposer and assignedProver is the same address, we named them differently because of back-ward compatible reasons. They ARE the same.

**Review by adaki2004 — APPROVED** [review]
LGTM but for some reason L2 Genesis tests seems to be failing - so disabled the auto-merge.

**Pull request merged**